wgsl/index.bs

```diff
@@ -1144,43 +1144,80 @@ A [=type rule=] has two parts:
     * Conditions on the other schematic parameters, if any.
     * Optionally, other static context.
 
-Each distinct type parameterization for a type rule is called an <dfn noexport>overload</dfn>.
-For example, [[#arithmetic-expr|unary negation]] (an expression of the form `-`|e|)
-has twelve overloads, because its type rules are parameterized by a type |T| that can be any of:
-* [=i32=]
-* [=vector|vec2&lt;i32&gt;=]
-* vec3&lt;i32&gt;
-* vec4&lt;i32&gt;
-* [=f32=]
-* [=vector|vec2&lt;f32&gt;=]
-* vec3&lt;f32&gt;
-* vec4&lt;f32&gt;
-* [=f16=]
-* [=vector|vec2&lt;f16&gt;=]
-* vec3&lt;f16&gt;
-* vec4&lt;f16&gt;
+Type rules may have <dfn noexport>type parameters</dfn> in their preconditions and conclusions.
+When a type rule's conclusion or preconditions contain type parameters,
+we say it is <dfn noexport>parameterized</dfn>.
+When they do not, we say the rule is <dfn noexport>fully elaborated</dfn>.
+We can make a fully elaborated type rule from a parameterized one
+by substituting a type for each of its type parameters,
+using the same type for all occurrences of a given parameter in the rule.
+An assignment of types to a rule's type parameters
+is called a <dfn noexport>substitution</dfn>.
+
+For example, here is the type rule for [[#logical-expr|logical negation]]
+(an expression of the form `!`|e|):
+
+<table class='data'>
+  <thead>
+    <tr><th>Precondition<th>Conclusion
+  </thead>
+  <tr algorithm="example boolean negation"><td>|e|: |T|<br>
+  |T| is bool or vec|N|&lt;bool&gt;
+  <td>`!`|e|`:` |T|
+</table>
+
+This is a parameterized rule, because it contains the type parameter |T|,
+which can represent any one of four types
+[=bool=], `vec2<bool>`, `vec3<bool>`, or `vec4<bool>`.
+Applying the substitution that maps |T| to `vec3<bool>`
+produces the fully elaborated type rule:
+
+<table class='data'>
+  <thead>
+    <tr><th>Precondition<th>Conclusion
+  </thead>
+  <tr algorithm="example2 boolean negation"><td>|e|`: vec3<bool>`<br>
+  <td>`!`|e|`: vec3<bool>`
+</table>
+
+Each fully elaborated rule we can produce from a parameterized rule
+by applying some substitution that meets the rule's other conditions
+is called an <dfn noexport>overload</dfn> of the parameterized rule.
+For example, the boolean negation rule has four overloads,
+because there are four possible ways to assign a type to its type parameter |T|.
+
+Note:
+In other words, a parameterized type rule provides the pattern 
+for a collection of fully elaborated type rules,
+each one produced by applying a different substitution to the parameterized rule.
 
 A <dfn noexport>type rule applies to a syntactic phrase</dfn> when:
 * The rule's conclusion matches a valid parse of the [=syntactic phrase=], and
 * The rule's preconditions are satisfied.
 
+A parameterized type rule applies to an expression
+if there exists a substitution producing a fully elaborated type rule
+that applies to the expression.
```

Comment:
I think these are now far enough from the definition that is worth turning them into links.

```suggestion
A [=parameterized=] type rule applies to an expression
if there exists a [=substitution=] producing a [=fully elaborated=] type rule
that applies to the expression.
```